synapse/handlers/room_summary.py

```diff
@@ -276,6 +277,8 @@ async def get_room_hierarchy(
         Returns:
             The JSON hierarchy dictionary.
         """
+        await self._request_ratelimiter.ratelimit(requester)
```

Comment:
I'm not sure if this is the correct ratelimiter to be using or not? It seems most of our other requests use a custom rate limiter instead of the request ratelimiter?

Reply:
Not sure here---might be worth another pair of eyes?

Reply:
It's not really appropriate, because `get_request_ratelimiter` is configured via the `rc_message` config settings, which are about the number of messages that can be sent.

Let's not introduce another set of config settings just for this, though. Either hardcode it for now or introduce some sort of "generic request" ratelimit setting.

Reply:
I went with the hard-coded settings for now (that are fairly high).

Reply:
>  It's not really appropriate, because get_request_ratelimiter is configured via the rc_message config settings, which are about the number of messages that can be sent.

Maybe a rename would help express this? (Messages as in to_device messages or events?) Not one for here though.